TweetMate/src/main/java/com/ybsystem/tweetmate/application/TweetMateApp.java

```diff
@@ -0,0 +1,94 @@
+package com.ybsystem.tweetmate.application;
+
+import android.annotation.SuppressLint;
+import android.app.Application;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.ybsystem.tweetmate.models.entities.Account;
+import com.ybsystem.tweetmate.models.entities.AppData;
+import com.ybsystem.tweetmate.models.entities.Via;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterUser;
+import com.ybsystem.tweetmate.databases.PrefEntity;
+
+import twitter4j.Twitter;
+import twitter4j.TwitterFactory;
+import twitter4j.auth.AccessToken;
+
+/**
+ * Application class of TweetMate
+ * Defines global variables used in the application
+ */
+public class TweetMateApp extends Application {
+
+    @SuppressLint("StaticFieldLeak")
+    private static TweetMateApp sTweetMateApp;
+
+    private AppCompatActivity mActivity;
+    private AppData mAppData;
+    private Twitter mTwitter;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        sTweetMateApp = this;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS2696)